### PR TITLE
Allergen Modulars

### PR DESCRIPTION
**Pull request by knharman**
This request contains the following changes:
- Refactored allergy-routes to share one handler
- fixed bug so the username always displays on all views
- altered the schema to disallow null values for allergens
- - also assumes all allergens unless told otherwise
- updated the seeds to match the schema

### CODE REVIEW

File: controllers/recipe-profile-routes.js [hunk 1]
```diff
@@ -41,10 +41,10 @@ router.get("/:id", (req, res) => {
       }
 
       const recipe = dbRecipeData.get({ plain: true });
-
       res.render("recipe-profile", {
         recipe,
         loggedIn: req.session.loggedIn,
+        name: recipe.user.username
```
Comment (knharman) [inline controllers/recipe-profile-routes.js:47]: fix for name missing in header

File: controllers/allergy-routes.js [hunk 2]
```diff
@@ -1,91 +1,60 @@
...
 const { Recipe, User, Vote, Allergy } = require("../models");
 
-const attributes = [ "id", "name", "description", "instructions", "ingredients", "user_id", 
-[ sequelize.literal( "(SELECT COUNT(*) FROM vote WHERE recipe.id = vote.recipe_id)" ), "vote_count",],];
-
-const include = [ { model: User, attributes: ["username"],}, {model: Allergy, as: 'allergies'} ]
-
-const catchError = (err) => { console.log(err); res.status(500).json(err); }
-
-router.get("/dairy", (req, res) => {
```
Comment (knharman) [inline controllers/allergy-routes.js:13]: matched on a param instead

File: public/javascript/allergen-filter.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ document.querySelector('#soy').addEventListener('click', soy);
 
 //nuts filter button
 async function nuts() {
-  document.location.replace('/allergy/nuts');
+  document.location.replace('/allergy/nut');
```
Comment (knharman) [inline public/javascript/allergen-filter.js:18]: this way it matches the database (not plural)

File: views/partials/single-recipe-description.handlebars [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <div class="recipe card m-3">
     <div class="card-body d-flex flex-column">
         
-        <h6 class="card-subtitle mb-2 text-muted">Submitted by: {{user.id}}</h6>
+        <h6 class="card-subtitle mb-2 text-muted">Submitted by: {{user.username}}</h6>
```
Comment (knharman) [inline views/partials/single-recipe-description.handlebars:4]: needed username, not Id to display user's kitchen

File: db/seeds.js [hunk 5]
```diff
@@ -65,7 +80,7 @@ const seed = async () => {
         ingredients: "walnut, almond, hazelnut, pecan, cashew pistachio, coconut"
     })
     await Allergy.create({
-        name: "Celiac Disease", 
+        name: "Celiac",
```
Comment (knharman) [inline db/seeds.js:62]: changed allergen to Celiac to make routing easier

**Review by heatherviolet — APPROVED** [review]
Changes look good

**Review by DrDano — APPROVED** [review]
Looks good, I see no code-breakers here!

**Pull request merged**